lib/req/request.ex

```diff
@@ -260,6 +260,43 @@ defmodule Req.Request do
     update_in(request.error_steps, &(steps ++ &1))
   end
 
+  @doc """
+  Runs the request steps in the request pipeline but does not execute them.
+
+  Accepts a `Req.Request` struct, or the same keyword list of options accepted
+  by `Req.new/1`.
+
+  Returns a `Req.Request` struct with all of its steps executed.
+
+  ## Examples
+
+      iex> prepared = Req.Request.prepare(method: :get, base_url: "http://www.example.com", params: [x: 1])
+      iex> prepared.url.host
+      "www.example.com"
+      iex> prepared.url.query
+      "x=1"
+  """
```

Comment:
`Req` calls functions from `Req.Request` but it shouldn't be the other way around. So let's not accept options, let's just accept a request struct.

```suggestion
  @doc """
  Prepares the request by running all of its request steps.

  ## Examples

      iex> req = Req.new(base_url: "https://httpbin.org", url: "/anything")
      iex> req = Req.Request.prepare(req)
      iex> to_string(req.url)
      "https://httpbin.org/anything"
  """
```